src/commonMain/kotlin/exchange.dydx.abacus/validator/TriggerOrdersInputValidator.kt

```diff
@@ -166,9 +169,84 @@ internal class TriggerOrdersInputValidator(
+        }
+
         return if (triggerErrors.size > 0) triggerErrors else null
     }
 
+    private fun validateStopLossTriggerToLiquidationPrice(
+        triggerOrder: Map<String, Any>,
+        position: Map<String, Any>,
+        tickSize: String,
+    ): List<Any>? {
+        val liquidationPrice = parser.asDouble(parser.value(position, "liquidationPrice.current"))
+        val triggerPrice = parser.asDouble(parser.value(triggerOrder, "price.triggerPrice"))
+
+        if (liquidationPrice == null || triggerPrice == null) {
+            return null
+        }
+
+        val type = parser.asString(triggerOrder["type"])
+        val side = parser.asString(triggerOrder["side"])
+
+        return when (requiredTriggerToLiquidationPrice(type, side)) {
+            RelativeToPrice.ABOVE -> {
+                if (triggerPrice <= liquidationPrice) {
+                    listOf(
+                        error(
+                            "ERROR",
+                            "SELL_TRIGGER_TOO_CLOSE_TO_LIQUIDATION_PRICE",
+                            listOf(TriggerOrdersInputField.stopLossPrice.rawValue),
+                            "APP.TRADE.MODIFY_TRIGGER_PRICE",
+                            "ERRORS.TRADE_BOX_TITLE.SELL_TRIGGER_TOO_CLOSE_TO_LIQUIDATION_PRICE",
+                            "ERRORS.TRADE_BOX.SELL_TRIGGER_TOO_CLOSE_TO_LIQUIDATION_PRICE",
+                            mapOf(
+                                "TRIGGER_PRICE_LIMIT" to mapOf(
+                                    "value" to liquidationPrice,
+                                    "format" to "price",
+                                    "tickSize" to tickSize,
+                                ),
+                            ),
+                        ),
+                    )
+                } else {
+                    null
+                }
+            }
+            RelativeToPrice.BELOW -> {
+                if (triggerPrice >= liquidationPrice) {
+                    listOf(
+                        error(
+                            "ERROR",
+                            "BUY_TRIGGER_TOO_CLOSE_TO_LIQUIDATION_PRICE",
+                            listOf(TriggerOrdersInputField.stopLossPrice.rawValue),
+                            "APP.TRADE.MODIFY_TRIGGER_PRICE",
+                            "ERRORS.TRADE_BOX_TITLE.BUY_TRIGGER_TOO_CLOSE_TO_LIQUIDATION_PRICE",
+                            "ERRORS.TRADE_BOX.BUY_TRIGGER_TOO_CLOSE_TO_LIQUIDATION_PRICE",
+                            mapOf(
+                                "TRIGGER_PRICE_LIMIT" to mapOf(
+                                    "value" to liquidationPrice,
+                                    "format" to "price",
+                                    "tickSize" to tickSize,
+                                ),
+                            ),
+                        ),
+                    )
+                } else {
+                    null
+                }
+            }
+            else -> null
```

Comment:
is this else case needed...? and if it is, i would again consider whether this is a valid state to be in and if we should just error out instead.

Reply:
with `enum` and `sealed` types, the compiler can enforce that we are checking every single case, so we shouldn't need an `else` case here.